### PR TITLE
CHANGE to use getDataForManager over extract api on storyStore

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: https://github.com/storybookjs/storybook/issues/11302

## What I did
I improved the `sb extract` cli command to use a better StoryStore api.

### CODE REVIEW

File: lib/cli/src/extract.ts [hunk 1]
```diff
@@ -17,28 +17,7 @@ const read = async (url: string) => {
...
-              framework: v.parameters.framework,
-              fileName: v.parameters.fileName,
-              docsOnly: v.parameters.docsOnly,
-            },
-          },
-        }),
-        {}
-      );
-      return JSON.stringify(result, null, 2);
+      return JSON.stringify(window.__STORYBOOK_STORY_STORE__.getDataForManager(), null, 2);
```
Comment (tmeasday) [inline lib/cli/src/extract.ts:20]: I thought we were (previously) being careful not to pull out *all* the parameters?

**Review by tmeasday — CHANGES_REQUESTED** [review]
I think maybe you want to do the earlier parameter filtering to `globalParameter`, each of the `kindParameters` and all the stories' individual parameters.

**Comment by tmeasday** [comment]
Although maybe we should add that code to the store itself so you call call `window.__STORYBOOK_STORY_STORE__.getStoriesJsonData` or the like?

File: lib/cli/src/extract.ts (hunk 1 above)
Comment (ndelangen) [reply lib/cli/src/extract.ts:20]: Yes... Ow right, parameters like `storysource` are going to make it in now..

**Comment by ndelangen** [comment]
maybe just `window.__STORYBOOK_STORY_STORE__.getStoriesforJSON()`?

File: lib/client-api/src/story_store.ts [hunk 2]
```diff
@@ -569,6 +570,35 @@ export default class StoryStore {
...
   };
 
+  getStoriesJsonData = () => {
+    const value = this.getDataForManager();
+    const allowed = ['fileName', 'docsOnly', 'framework', '__id'];
+
+    return {
+      v: 2,
+      globalParameters: pick(value.globalParameters, allowed),
+      kindParameters: Object.entries(value.kindParameters).reduce(
```
Comment (shilman) [inline lib/client-api/src/story_store.ts:585]: `mapValues`?

File: lib/client-api/src/story_store.ts [hunk 3]
```diff
@@ -569,6 +570,35 @@ export default class StoryStore {
     };
   };
 
+  getStoriesJsonData = () => {
+    const value = this.getDataForManager();
+    const allowed = ['fileName', 'docsOnly', 'framework', '__id'];
```
Comment (shilman) [inline lib/client-api/src/story_store.ts:580]: `__isArgsStory`?

**Review by shilman — COMMENTED** [review]
LGTM with a couple suggestions

File: lib/client-api/src/story_store.ts (hunk 3 above)
Comment (tmeasday) [reply lib/client-api/src/story_store.ts:580]: Do we need to know that for the sidebar?

**Review by tmeasday — APPROVED** [review]
LGTM now! I agree `mapValues` is better. Also maybe we can type it?

**Comment by ndelangen** [comment]
Cool I added a test as well

**Pull request merged**